alphastats/gui/pages/03_Preprocessing.py

```diff
@@ -1,16 +1,65 @@
 import streamlit as st
+import pandas as pd
 
-try:
-    from alphastats.gui.utils.preprocessing_helper import main_preprocessing
-    from alphastats.gui.utils.ui_helper import sidebar_info
 
-except ModuleNotFoundError:
-    from utils.preprocessing_helper import main_preprocessing
-    from utils.ui_helper import sidebar_info
+from alphastats.gui.utils.preprocessing_helper import (
+    draw_workflow,
+    configure_preprocessing,
+    display_preprocessing_info,
+    update_workflow,
+    run_preprocessing,
+    reset_preprocessing,
+    PREPROCESSING_STEPS,
+)
+from alphastats.gui.utils.ui_helper import sidebar_info
 
 
 sidebar_info()
 
+if "workflow" not in st.session_state:
+    st.session_state["workflow"] = [
+        PREPROCESSING_STEPS.REMOVE_CONTAMINATONS["key"],
+        PREPROCESSING_STEPS.SUBSET["key"],
+        PREPROCESSING_STEPS.LOG2_TRANSFORM["key"],
+    ]
+
 st.markdown("### Preprocessing")
+c1, c2 = st.columns([1, 1])
+
+with c2:
+    if "dataset" not in st.session_state:
+        settings = {k: True for k in st.session_state.workflow}
```

Comment:
seems a bit hacky .. 
the explicit way would be
```
def update_workflow(
    settings: dict, show_all: bool = False
):
    new_workflow = [
        key
        for key, setting in settings.items()
        if show_all or setting
    ]
    return new_workflow
```

Reply:
https://github.com/MannLabs/alphapeptstats/pull/308/commits/e398d76ada357c5e40276a1e0b1bea6432b1b170

It was entirely unnecessary to even create settings if no dataset is present.